src/python_testing/TC_MCORE_FS_1_2.py

```diff
@@ -0,0 +1,185 @@
+                 TestStep(6, "TH_FSA initiates commissions of TH_SED_L by sending the OpenCommissioningWindow command to the Administrator Commissioning Cluster on the endpoint with the uniqueID matching that of TH_SED_L."),
+                 TestStep(7, "TH_FSA completes commissioning of TH_SED_L using the Enhanced Commissioning Method.")]
+        return steps
+
+    @async_test_body
+    async def test_TC_MCORE_FS_1_2(self):
+        self.is_ci = self.check_pics('PICS_SDK_CI_ONLY')
+        min_report_interval_sec = self.user_params.get("min_report_interval_sec", 0)
+        max_report_interval_sec = self.user_params.get("max_report_interval_sec", 10 * 60)
+        timeout_delay_sec = self.user_params.get("timeout_delay_sec", max_report_interval_sec * 2)
+
+        self.step(1)
+
+        # Subscribe to the UniqueIDs
+        unique_id_queue = queue.Queue()
+        subscription_contents = [
+            (Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID)  # On all endpoints
+        ]
+        sub = await self.default_controller.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=subscription_contents,
+            reportInterval=(min_report_interval_sec, max_report_interval_sec),
+            keepSubscriptions=False
+        )
+        attribute_handler = AttributeChangeAccumulator(
+            name=self.default_controller.name, expected_attribute=Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID, output=unique_id_queue)
+        sub.SetAttributeUpdateCallback(attribute_handler)
+        sub_handlers.append(attribute_handler)
+
+        logging.info("Waiting for First BridgedDeviceBasicInformation.")
+        start_time = time.time()
+        elapsed = 0
+        time_remaining = timeout_delay_sec
+
+        th_sed_dut_bdbi_endpoint = -1
+        th_sed_dut_unique_id = -1
+
+        while time_remaining > 0 and th_sed_dut_bdbi_endpoint < 0:
+            try:
+                item = unique_id_queue.get(block=True, timeout=time_remaining)
+                client_name, endpoint, attribute, value = item['name'], item['endpoint'], item['attribute'], item['value']
+
+                # Record arrival of an expected subscription change when seen
+                if attribute == Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID:
+                    th_sed_dut_bdbi_endpoint = endpoint
+                    th_sed_dut_unique_id = value
+
+            except queue.Empty:
+                # No error, we update timeouts and keep going
+                pass
+
+            elapsed = time.time() - start_time
+            time_remaining = timeout_delay_sec - elapsed
+
+        logging.info("Found BDBI with UniqueID (%d) on endpoint %d." % th_sed_dut_unique_id, th_sed_dut_bdbi_endpoint)
+
+        self.step(2)
+
+        self.sync_passcode = 20202024
+        self.th_sed_dut_discriminator = 2222
+        # min commissioning timeout is 3*60 seconds, so use that even though the command said 30.
```

Comment:
I'm not sure this comment makes sense in this context.

Spec Q - I know this is from my code, but what's actually supposed to happen here if the commissioning timeout is 30? That's too low for a commissioning window.

Reply:
I'm not sure where you get the 30? There is some stuff around FS ICDs that used to say 30, but I don't think there was anything around FS commissioning.

Reply:
Different question - let's talk offline.